### DISCUSSION TITLE
Markdoc language server: How to config preview?

### DISCUSSION THREAD

**Discussion by amozoss** [markdoc/markdoc#444] [Q&A]
I tried setting the `preview` in my `markdoc.config.json`. I tried setting it to `true` and my local nextjs server. The globe button appears
![Screenshot 2023-09-08 at 1 13 08 PM](https://github.com/markdoc/markdoc/assets/1147264/39d3d398-43cd-499f-8532-c6389a58c8e0)

However, nothing shows up in the preview.

`markdoc.config.json`
```
{
  "id": "storj-docs",
  "path": "app",
  "preview": "http://localhost:3000",
  "schema": {
    "path": "src/markdoc/config.mjs",
    "type": "esm",
    "property": "default",
    "watch": true
  }
} 
```

I ran the extension locally and this is the error it is throwing.

```
stack trace: Error: Unhandled method markdoc.renderPreview
	at handleResponse (/Users/dan/test/language-server/dist/client/index.js:4518:40)
	at processMessageQueue (/Users/dan/test/language-server/dist/client/index.js:4356:13)
	at Immediate.<anonymous> (/Users/dan/test/language-server/dist/client/index.js:4342:11)
	at process.processImmediate (node:internal/timers:466:21)
	at process.callbackTrampoline (node:internal/async_hooks:130:17)
```

How do I configure the preview?

**Comment by rpaul-stripe** [comment]
This feature is still somewhat experimental. At the present time, it relies on a `markdoc.renderPreview` command being supported in the language server. Given that the approach for rendering content can differ pretty widely between Markdoc projects, the language server does not provide an implementation of this feature by default — it has to be implemented in a custom language server plugin.

The intent for this particular feature was that individual frameworks that make custom variants of the language server would take advantage of it to support framework-specific previews that rely on whatever routing logic they use to show a preview for the file that has focus. In practice, I don't think anyone has done that yet.

We don't have any documentation on how to do this just yet, but we'll publish some soon.

**Reply by amozoss** [reply]
Thanks for the response. Excited for this option to be available, as we have non-technical folks who are currently using the github web based ui to edit our markdoc markdown, but it'd be awesome to give them an easy way to preview in vscode with the markdoc specific things rendered too.

So if I understand what you're saying, a concrete example would basically be, `markdoc/nextjs` would implement the `renderPreview`?
